### PR TITLE
fix: Remove notification list Show More button and design fixes

### PR DESCRIPTION
**Pull request by olerichter00**
The type of this PR is: **Fix**

This PR solves [ONYX-688]

### Description

This pull request removes the "Show More" button from the notification list and includes some design fixes:

-  Removes the bottom border of the notification list.
-  Adjusts the maximum height of the notification list to prevent over-scrolling.

Screenshot 2024-02-13 at 15 12 18

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ONYX-688]: https://artsyproduct.atlassian.net/browse/ONYX-688?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Components/Notifications/NotificationsList.tsx [hunk 1]
```diff
@@ -90,37 +87,9 @@ const NotificationsList: React.FC<NotificationsListProps> = ({
       if (err) console.error(err)
 
       setLoading(false)
-
-      // Change pagination type to "infinite" when "show more" button was pressed
-      if (paginationType === "showMoreButton") {
-        setCurrentPaginationType("infinite")
-      }
```
Comment (olerichter00) [inline src/Components/Notifications/NotificationsList.tsx:97]: We never show the button and always use infinite scroll.

File: src/Components/Notifications/NotificationsWrapper.tsx [hunk 2]
```diff
@@ -8,7 +8,7 @@ import { MarkAllAsReadPanelProps } from "Components/Notifications/MarkAllAsReadP
 const DROPDOWN_HEADER_HEIGHT = 113
 const VERTICAL_OFFSET = 10
 const DROPDOWN_CONTENT_HEIGHT =
-  DROPDOWN_HEADER_HEIGHT + DESKTOP_NAV_BAR_TOP_TIER_HEIGHT + VERTICAL_OFFSET
+  DROPDOWN_HEADER_HEIGHT + DESKTOP_NAV_BAR_HEIGHT + VERTICAL_OFFSET
```
Comment (olerichter00) [inline src/Components/Notifications/NotificationsWrapper.tsx:11]: Before this change, the maximum height was too large, so it was possible to scroll out of the container.

**Review by nickskalkin — APPROVED** [review]
👍

**Review by tam-kis — APPROVED** [review]

File: src/Components/Notifications/NotificationsWrapper.tsx (hunk 2 above)
Comment (tam-kis) [reply src/Components/Notifications/NotificationsWrapper.tsx:11]: Makes sense, we made some changes here so it could've broke it.

**Pull request merged**